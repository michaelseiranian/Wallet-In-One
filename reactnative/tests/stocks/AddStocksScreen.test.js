import React from "react";
import { render, screen, fireEvent, act, waitFor} from '@testing-library/react-native';
import PlaidComponent from "../../screens/stocks/AddStocksScreen";
import { PlaidEnvironments, Configuration, PlaidApi,  } from 'plaid';
import { describe } from "@jest/globals";

/**
 * Tests for the Plaid SDK
 */
const configuration = new Configuration({
  basePath: PlaidEnvironments.sandbox,
  baseOptions: {
    headers: {
      'PLAID-CLIENT-ID': '63ef90fc73e3070014496336',
      'PLAID-SECRET': 'a57f2537ac53e9842da752b987bb5b',
    },
  },
});

const client = new PlaidApi(configuration);
prods = ['investments', 'transactions'];

describe('<PlaidComponent />', () => {

    beforeEach(() => {
        global.fetch =  jest.fn( async (url) => {
            const request = await client.linkTokenCreate({
                products: prods,
                client_name: "KCL",
                country_codes: ['GB'],
                language: 'en',
                user: {
                    client_user_id: '1',
                  },
            })
            if(url.includes('/stocks/initiate_plaid_link/')){
            return Promise.resolve({
                status: 200, json: () => ({ link_token: request.data.link_token })
            })
            }
        })
      })

    afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })
    
    it('addStocksScreen snapshot test', async () => {
        const addStocksScreen = render(<PlaidComponent />);
        expect(addStocksScreen).toMatchSnapshot();
    }
    )
})