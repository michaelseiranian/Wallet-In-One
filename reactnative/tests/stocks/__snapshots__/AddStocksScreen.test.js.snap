// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaidComponent /> addStocksScreen snapshot test 1`] = `
[
  <View
    style={
      [
        {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={true}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        {
          "uri": "https://cdn.plaid.com/link/v2/stable/link.html?isWebview=true&token=",
        }
      }
      style={
        [
          {
            "flex": 1,
            "overflow": "hidden",
          },
          {
            "backgroundColor": "#ffffff",
          },
          {
            "flex": 1,
            "marginTop": 54,
          },
        ]
      }
      textInteractionEnabled={true}
      useSharedProcessPool={true}
    />
  </View>,
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      />
    </View>
  </View>,
]
`;
